### ISSUE TITLE
Invalid methodID for MFA-enabled user

### ISSUE THREAD

**Issue by spsaucier** [FusionAuth/fusionauth-issues#1627]
### Description 

After I go through `api/login` with credentials of a user who has 2FA enabled, I get this response:

```
{"methods":[{"id":"4ZF7","lastUsed":true,"method":"sms","mobilePhone":"17062259155"}],"twoFactorId":"ETCK_CQPo5ARy7SvrWC7vd4rC0ilbNRSc52pdE0rDd0"}
```

I then immediately grab the method ID and `twoFactorId` from that and request that FA send a code to that two factor method, like so:

```
curl -v
-X POST 'https://fa.capital.dev.tranwall.net/api/two-factor/send/ETCK_CQPo5ARy7SvrWC7vd4rC0ilbNRSc52pdE0rDd0'
-H 'Cookie: access_token=eyJhb...; refresh_token=Zzanp...'
-H 'Accept: application/json, text/plain, */*'
-H 'Content-Type: application/json'
-d '{"methodId":"4ZF7"}'
```

Unfortunately, this responds with a `400`:

```
{"fieldErrors":{"methodId":[{"code":"[invalid]methodId","message":"The [methodId] is not valid. No two-factor method with this Id was found enabled for the user."}]}}
```

### Affects versions
Running version `1.34.1`

The methodId should not be invalid, since it is given to me by FA in the first place. I have re-confirmed that the method ID is identical in FusionAuth.

### Steps to reproduce
Unsure - it affects only a portion of users, but we have seen it with several users who have MFA enabled.

### Expected behavior
The method ID that is sent in the login response should be acceptable to `two-factor/send/${twoFactorId}`.

### Platform
 - Device: Desktop
 - OS: macOS
 - Browser + version: Chrome 98, Android, iOS (latest)
 - Database: Hosted FusionAuth

**Comment by spsaucier** [comment]
Is it possible that including the `Cookie` header is confusing FusionAuth?

**Comment by mooreds** [comment]
@spsaucier It shouldn't be, but if you remove it do things work the way you'd expect them to?

**Comment by spsaucier** [comment]
Actually, no, the cookie header does not affect this.

Trying this again now, it seems as if there is a race condition -- I get the new `twoFactorId` and immediately send it off to trigger sending the SMS (in my case). Maybe this `twoFactorId` is not yet accessible to the `send` endpoint?

**Comment by spsaucier** [comment]
This case probably exists particularly when the app was logged in as a different user, they log out (and don't clear cookies), and log in as another user.

**Comment by robotdan** [comment]
This would be the expected behavior if the `access_token` cookie sent on the request belongs to a different user than owns the `twoFactorMethodId` provided on the URL segment.

If I understand you correctly @spsaucier this was the case?

**Comment by spsaucier** [comment]
Yes, that was the case. We are now clearing cookies before making this request.

**Comment by mooreds** [comment]
Updated the label as we might want to make that clearer in the docs.
